src/main/resources/SLING-INF/nodetypes/nodetypes.cnd

```diff
@@ -33,5 +33,9 @@
 	- pant:otherNames (string) multiple
 
 // Base type for modules
-[pant:module] > nt:unstructured
-	- sling:resourceType (string) = 'pantheon/modules' mandatory autocreated
\ No newline at end of file
+[pant:module] > nt:unstructured, sling:Folder
+	- sling:resourceType (string) = 'pantheon/modules' mandatory autocreated
+    + cachedContent (pant:moduleMetadata) = pant:moduleMetadata mandatory autocreated
```

Comment:
So the bottom-line motivation behind these .cnd changes is that I want the cachedContent node to get autocreated. Here's why: it's possible for a logged-in user to upload a module but NOT preview it. Then, later, an anonymous user could come along and request a preview. Our asciidoctor code would run, but since it's in the context of the anonymous user, it doesn't have permission to write the result to the cache node. It *could* deliver the html output and then drop it on the floor instead of caching it, but I think that's a tad on the stupid side.

I see two alternative approaches:
1. Allow the anonymous user to create child nodes
2. Ensure the cache node is already available, and grant anonymous the ability to modify its properties

I like option 2 because it's narrower in scope and let's say "less insecure." Either way we are exposing a potential attack vector for a bad actor, but my hope is that polluting a single node with a bunch of nonsense properties is "less bad" then polluting the structure with a bunch of nonsense nodes.

At any rate, there's probably a better approach from a security perspective but this suffices for our current stage of just getting the software off the ground.

Reply:
ah, I see... I wonder if there's a way we can create the cached content node as some sort of 'system only' node, where the system itself would be the only one with permissions to modify it (or even read from it?). This node is only going to be manipulated by the app, either by creating the cached content or reading it to deliver it. No user (other than admins) should have direct access to it.

Reply:
I agree, that would be best. I don't currently know how to elevate the permissions of code running in a particular context. But as the project matures I think that would be a valuable technique to have in our bag of tricks.